File: src/android/GetSocialPluginProxy.java
package com.getsocialsdk.cordova.plugin;

import android.content.Context;
import android.graphics.Bitmap;

import org.apache.cordova.CallbackContext;
import org.apache.cordova.PluginResult;
import org.json.JSONObject;

import java.util.List;

import im.getsocial.sdk.core.plugins.InvitePlugin;

/**
 * Created by bevzhome on 03.02.16.
 */
public class GetSocialPluginProxy extends InvitePlugin {
    private CallbackContext callbackContext;
    boolean isAvailableForDevice = false;
    InviteFriendsObserver inviteFriendsObserver = null;

    public GetSocialPluginProxy(CallbackContext callbackContext, boolean isAvailableForDevice) {
        this.callbackContext = callbackContext;
        this.isAvailableForDevice = isAvailableForDevice;
    }

    public boolean onComplete(String s, List<String> list) {
        if (inviteFriendsObserver == null)
            return false;

        inviteFriendsObserver.onComplete(s, list);

        return true;
    }

    public boolean onCancel() {
        if (inviteFriendsObserver == null)
            return false;

        inviteFriendsObserver.onCancel();

        return true;
    }

    public boolean onError() {
        if (inviteFriendsObserver == null)
            return false;

        inviteFriendsObserver.onError(new Exception());

        return true;
    }
    public boolean onClicked() {
        if (inviteFriendsObserver == null)
            return false;

        inviteFriendsObserver.onClicked();

        return true;
    }

    @Override
    public void inviteFriends(Context context, String s, String s1, String s2, Bitmap bitmap, InviteFriendsObserver inviteFriendsObserver) {
        this.inviteFriendsObserver = inviteFriendsObserver;

        JSONObject result = new JSONObject();
        try {
            result.put("action", "inviteFriends");
            result.put("subject", s);
            result.put("text", s1);
            result.put("referralDataUrl", s2);
            result.put("image", GetSocialUtil.convertBitmapToBase64(bitmap));
        } catch (Exception e) {}

        PluginResult pluginResult = new PluginResult(PluginResult.Status.OK, result);
        pluginResult.setKeepCallback(true);
        callbackContext.sendPluginResult(pluginResult);
    }

    public void emulateInviteFriends() {
        JSONObject result = new JSONObject();
        try {
            result.put("action", "inviteFriends");
            result.put("s1", "subject");
        } catch (Exception e) {}

        PluginResult pluginResult = new PluginResult(PluginResult.Status.OK, result);
        pluginResult.setKeepCallback(true);
        callbackContext.sendPluginResult(pluginResult);
    }

    public void initCallbackId(String callbackId) {
        JSONObject result = new JSONObject();
        try {
            result.put("action", "initCallbackId");
            result.put("callbackId", callbackId);
        } catch (Exception e) {}

        PluginResult pluginResult = new PluginResult(PluginResult.Status.OK, result);
        pluginResult.setKeepCallback(true);
        callbackContext.sendPluginResult(pluginResult);
    }

    @Override
    public boolean isAvailableForDevice(Context context) {
        return isAvailableForDevice;
    }


}

File: src/ios/GetSocialPluginProxy.m
//
//  GetSocialPluginProxy.m
//  GetSocial
//
//  Created by Igor on 2/8/16.
//
//

#import "GetSocialPluginProxy.h"

@implementation GetSocialPluginProxy

-(id) initWithCommandDelegate:(id<CDVCommandDelegate>) commandDelegate callbackId:(NSString*) callbackId available:(BOOL) availableForDeviceStatus {
    self = [super init];
    if (self) {
        [self setAvailableForDeviceStatus:availableForDeviceStatus];
        [self setCallbackId:callbackId];
        [self setCommandDelegate:commandDelegate];
    }
    return self;
}

-(BOOL) onComplete:(NSDictionary*) dict {
    if([self successCallback] == nil)
        return false;
    
    self.successCallback(dict);
    return true;
}

-(BOOL) onCancel {
    if([self cancelCallback] == nil)
        return false;
    
    self.cancelCallback();
    return true;
}

-(BOOL) onError {
    if([self errorCallback] == nil)
        return false;
    
    self.errorCallback([[NSError alloc] init]);
    return true;
}

-(void) inviteFriendsWithSubject:(NSString *)subject text:(NSString *)text referralDataUrl:(NSString *)referralDataUrl image:(UIImage *)image onViewController:(UIViewController *)onViewController success:(GetSocialInviteSuccessCallback)successCallback cancel:(GetSocialCancelCallback)cancelCallback error:(GetSocialErrorCallback)errorCallback {
    
    [self setSuccessCallback:successCallback];
    [self setCancelCallback:cancelCallback];
    [self setErrorCallback:errorCallback];
    
    NSMutableDictionary *property = [[NSMutableDictionary alloc] init];
    
    property[@"action"] = @"inviteFriends";
    property[@"subject"] = subject;
    property[@"text"] = text;
    property[@"referralDataUrl"] = referralDataUrl;
    property[@"image"] = [self convertUIImageToBase64:image];
    
    CDVPluginResult *result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:property];
    [result setKeepCallbackAsBool:YES];
    [self sentResult:result];
}

-(void) sentResult:(CDVPluginResult*) pluginResult  {
    if (self.commandDelegate != nil) {
        [self.commandDelegate sendPluginResult:pluginResult callbackId:[self callbackId]];
    }
}

-(NSString*) convertUIImageToBase64:(UIImage*) image {
    return @"";
}

-(void) initializeCallbackId {
    NSDictionary *property = @{@"action":@"initCallbackId", @"callbackId":[self callbackId]};
    CDVPluginResult* result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:property];
    [result setKeepCallbackAsBool:YES];
    [self sentResult:result];
}

-(BOOL) isAvailableForDevice {
    return self.availableForDeviceStatus;
}

@end
